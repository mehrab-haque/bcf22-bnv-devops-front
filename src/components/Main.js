import react, {useEffect} from 'react'
import {showToast} from "../App";

const Main=()=>{

    useEffect(()=>{
        showToast("Hello World")
        console.log("hi")
    },[])

    return(
        <div>
            Hello World
        </div>
    )
}

export default Main