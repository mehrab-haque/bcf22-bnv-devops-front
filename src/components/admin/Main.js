import react, {useEffect, useState} from 'react'
import {showToast} from "../../App";
import Dashboard from "./Dashboard";
import Auth from "./Auth";

var getToken

const Main=()=>{

    const [isAuth,setAuth]=useState(false)
    const [token,setToken]=useState(null)

    getToken=()=>{
        return token
    }

    const login=()=>{
        setAuth(true)
    }

    const logout=()=>{
        setAuth(false)
    }

    return(
        <div>
            {
                isAuth?(
                    <Dashboard logout={logout}/>
                ):(
                    <Auth login={login} setToken={setToken}/>
                )
            }
        </div>
    )
}

export default Main
export {getToken}