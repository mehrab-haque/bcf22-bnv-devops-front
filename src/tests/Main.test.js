import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Main from "../components/admin/Main";

describe("Component Main Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Main/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
