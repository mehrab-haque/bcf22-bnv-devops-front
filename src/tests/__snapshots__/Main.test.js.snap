// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Main Render renders without crashing given the required props 1`] = `
<div>
  <Auth
    login={[Function]}
    setToken={[Function]}
  />
</div>
`;
